### ISSUE TITLE
Is it possible to open the reply "window" in another terminal instance (or tmux window etc.?)

### ISSUE THREAD

**Issue by sanjayankur31** [neomutt/neomutt#2713]
I realise this is a very niche use case, but I thought I'd ask anyway.

I sometimes take quite a long time to reply to e-mails. First, second, third drafts, and I tend to leave the reply screen open and go back to it later etc. to improve my response. Now, if I have this reply screen open, I either need to save/postpone the message or open another neomutt instance to read the e-mails that may have come in. So, I wonder if there's a way to perhaps add a new command  to "reply in a new window" and let neomutt work away where it is? It doesn't have to be a new terminal instance/tab. I use tmux, for example, so if I can even "reply in a new tmux window", that'll work too.

I tried something like:
```
set editor="tmux new-window vimx %s"
```
This does open a new tmux window for me, but the file that vim opens there is completely empty.
I couldn't think of anything else to try.

Cheers,

**Comment by flatcap** [comment]
tl;dr No :-)

Upstream Mutt has a feature "background compose" which allows you to have multiple compose sessions running simultaneously.
I haven't merged it yet because it's a big change and it is very ugly -- it backs up and restores hundreds of config variables.
I'm still trying to refactor the code to make it cleaner, but I simply don't have the time.

> `set editor="tmux new-window vimx %s"`

That nearly works, but it suffers from the mailcap problem.
`tmux new-window` queues the request and returns immediately.
NeoMutt then carries on with its composing (and the tmp file gets deleted).
By the time `vim` is run, the file's gone.

You **can't** background editors in NeoMutt, but you can get the editor in a separate tmux window.
I'm not a tmux expert, but this works:
(I couldn't work out the escaping to make it into a single script)

**neomutt.rc**
```
set editor='one.sh'
```

**one.sh**
```sh
#!/bin/bash
NAME="neomutt-$RANDOM"
tmux new-window -n "$NAME" two.sh "$NAME" "$@"
tmux wait "$NAME"
```

**two.sh**
```sh
#!/bin/bash
NAME="$1"
shift
vi "$@"
tmux wait -S "$NAME"
```

**Comment by jindraj** [comment]
I only merged scripts into one. I only changed new window to split so I could better see the result.

```sh
#!/bin/bash
RND=$RANDOM
tmux split-window -h "$EDITOR $@; tmux wait -S $RND"
tmux wait $RND
```

**Comment by flatcap** [comment]
The problem I couldn't fix was quoting the `$@` to preserve filenames containing whitespace.
For NeoMutt that isn't a problem.

**Comment by jindraj** [comment]
> The problem I couldn't fix was quoting the `$@` to preserve filenames containing whitespace.

This seems to work also
```sh
tmux split-window -h "$EDITOR \"$@\"; tmux wait -S $RND"
```

> For NeoMutt that isn't a problem.

I know it's not a problem, but it can become easily…
export TMPDIR="/tmp/directory with spaces/"

**Comment by sanjayankur31** [comment]
Thanks for the hints @flatcap @jindraj .

This is what I came up with. It's a bit roundabout, but gets it done:

```
# neomutt-compose.sh
RNDNEW="$(date +%H%M)"
tmux new-window -ad -n "neomutt-$RNDNEW" neomutt && tmux swap-window -d -t +1 && vimx "$@"
```

So, this opens a new neomutt instance for me in a new tmux window (lets call it the reader instance), swaps it with the current one (which we can call the reply instance), and then opens the editor in the reply instance. 

The swapping bit is optional, I just like to have a neomutt reader instance in a particular place in tmux as a matter of habit, and one can remove the `-d` flag there if the don't want the reply instance to be focused---sort of put it in the background.

Now that I think of it, this just comes down to me being lazy. I could just have opened a new instance of neomutt manually ias I've been doing now :laughing:

**Comment by sanjayankur31** [comment]
Ah, caveat (as expected): this opens a new "reply" instance each time one hits "e" to edit the mail again on the confirmation screen too. Not a deal-breaker. From only one neomutt instance, I've just gone to having quite a few---which works too XD
